### PR TITLE
skip,metamorphic: break dependency on `metamorphic` from `skip`

### PR DESCRIPTION
**Pull request by rickystewart**
It would be nice to have a way to clearly determine whether any one package uses metamorphic constants, however that's difficult since `pkg/testutils/skip` (which many test packages use) depends on `metamorphic`. This PR breaks that dependency, so we can query the dependency graph to the `metamorphic` dependency to determine exactly which packages use metamorphic constants. Instead, we have `skip` depend on a new package `metamorphicutil` which is very restricted in use. `metamorphic` can also do setup in its `init()` function that we know for sure is not needed unless metamorphic constants are specifically requested.

Epic: none
Release note: None

### CODE REVIEW

File: pkg/util/metamorphic/metamorphicutil/is_metamorphic.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package metamorphicutil
```
Comment (srosenberg) [inline pkg/util/metamorphic/metamorphicutil/is_metamorphic.go:11]: Nit: `util.metamorphic.metamorphicutil` is kinda redundant... if we don't immediately expect other additions to this package, maybe `buildflags` is a better name?

**Review by srosenberg — APPROVED** [review]

File: pkg/util/metamorphic/metamorphicutil/is_metamorphic.go (hunk 1 above)
Comment (rickystewart) [reply pkg/util/metamorphic/metamorphicutil/is_metamorphic.go:11]: I'm not 100% clear on the details, but I do expect other additions to this package. I think that we will probably add some sort of `map[string]interface{}` as a "catalog" of the metamorphic variables. So `buildflags` I don't think will be appropriate. Won't be hard to change the name later if a better one is found while we're adding the new features.

**Comment by rickystewart** [comment]
TFTR!

bors r=srosenberg

**Comment by rickystewart** [comment]
bors r=srosenberg

**Pull request merged**